### DISCUSSION TITLE
Error when defining a relationship on SQL Server

### DISCUSSION THREAD

**Discussion by wodraeve** [Budibase/budibase#8544] [Help] [answered]
Data Source: (Azure) SQL Server

Steps:

1. create connection. No special things here, regular sql authentication. **schema is not dbo!**
2. table fetch works. Fetching two tables. 1 is "employees"; 2 is "employee dates". In SQL Server a FK relationship is defined via [employee_dates].[employee_id]  to [ employees].[id]
3. records are showing up fine in the data preview
4. creating a relationship in budibase:

- go to the data configuration
- click on 'Define relationships' (none defined at this point)
Details:
- 'One'-type relationship
- from 'employees' primary key 'id'
- to 'employee_dates' foreign key 'employee_id'
No changes made to the suggested column names.
- click 'Save'

"Datasource updated successfully".
- Navigating to 'employees' generates error.

`RequestError: Invalid object name 'employee_employment_dates'`

The records in the table no longer are visible.

Any thoughts? This should be a straightforward action (actually I feel SQL Server foreign key relationships should be automatically defined as relationships in budibase, but I can recreate them)

**Comment by melohagan** [comment]
Hey @wodraeve 

Would you be able to provide me with the CREATE script for those tables please; then I can have a go at reproducing.

**Comment by koneubauer** [comment, marked as answer]
Hello!

We have the same issue. We are using version 2.1.24
When creating a table in a schema that isn't dbo, creating relationships trigers the same error as mentioned above.
In our case, the structure of the tables doesn't matter.

**Reply by melohagan** [reply]
Hey @koneubauer 

There has been recent work done around the relationships. Have you tried upgrading to **2.3.16** ?

**Reply by koneubauer** [reply]
Hey, we haven't upgraded yet. We'll try and keep you posted.

**Reply by koneubauer** [reply]
Works in the newer version, thanks.
